client/src/components/App.jsx

```diff
@@ -1,17 +1,61 @@
+/* eslint-disable camelcase */
 import React from 'react';
+import axios from 'axios';
 import './App.css';
 class App extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      productId: 47425,
+      displayProduct: [],
+    };
+    this.formatBody = this.formatBody.bind(this);
+
+  }
+
+  formatBody(method, apiRoute, params, data) {
+
+    let bodyObj = {
+      method: method,
+      url: `https://app-hrsei-api.herokuapp.com/api/fec2/hr-rpp${apiRoute}`,
+      data: data,
+      params: params,
```

Comment:
If **data** and **params** are empty for some call, we can put some **default value** for them instead of **undefined**. Ex: **_data: data || {}_** or we can also put in the method signature. **_formatBody(method, apiRoute, params, data={})_**

Reply:
Agreed.  I prefer to put them  as defaults in the actual function to minimize user error.  Sound Ok?

Reply:
Sounds good to me.